### PR TITLE
Support custom objectives and optimizers

### PR DESCRIPTION
**Pull request by tzachar**
Support custom objectives and optimizers by adding custom_objects to optimizers.get and objectives.get

### CODE REVIEW

**Comment by dbonadiman** [comment]
Can you do a simple example or use case about that feature?

**Comment by tzachar** [comment]
Sure. I use the following for quantile regression. y_true holds the np.vstack([y] * nb_samples).T
def batch_q_loss(y_true, y_pred):
   import theano.tensor as T
   quantiles=np.arange(0.05, 1., 0.05)
   diff = y_true - y_pred
   return T.mean(0.5 * (2 * quantiles - 1) * diff + 0.5 * abs(diff))

**Comment by dbonadiman** [comment]
Just to understand what is the problem in plugging that in place of the loss during compilation? 
```python
model.compile(loss= batch_q_loss, optimizer='sgd')
```

**Comment by tzachar** [comment]
This will not work when serializing the model.
compiled is called there without the user's intervention, and tries to
recreate the loss function.

File: keras/models.py [hunk 1]
```diff
@@ -141,12 +141,21 @@ def model_from_config(config, custom_objects={}):
 
         optimizer_params = dict([(k, v) for k, v in config.get('optimizer').items()])
         optimizer_name = optimizer_params.pop('name')
-        optimizer = optimizers.get(optimizer_name, optimizer_params)
+        optimizer = optimizers.get(optimizer_name, kwargs=optimizer_params, custom_objects=custom_objects)
 
         if model_name == 'Sequential':
-            model.compile(loss=loss, optimizer=optimizer, class_mode=class_mode, theano_mode=theano_mode)
```
Comment (EderSantana) [inline keras/models.py]: it was ok before, why that?

**Comment by EderSantana** [comment]
custom objectives (optimizers) are already supported, you just have to write you own function (class) and pass them as parameters to compile

**Comment by tzachar** [comment]
Custom objectives are not supported.
Check the following unit test, which throws an exception on the current code:

```python
from keras.models import Sequential, model_from_json
from keras.layers.core import Dense
from keras.optimizers import SGD


def my_mse(y, pred):
    import theano.tensor as T
    return T.mean((y - pred) ** 2)


def create_model():
    model = Sequential()
    model.add(Dense(100, input_shape=(100,)))
    return model

model = create_model()
sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
model.compile(loss=my_mse, optimizer=sgd)

open('/tmp/json_model', 'w').write(model.to_json())

model = model_from_json(open('/tmp/json_model').read(), custom_objects={'my_mse': my_mse})
```

**Comment by fchollet** [comment]
Closing outdated PRs following the release of Keras 1.0.
